### ISSUE TITLE
High CPU usage

### ISSUE THREAD

**Issue by mattno** [terrymacdonald/DisplayMagician#110]
**Describe the bug**
When starting a game using Display Magician, CPU usage goes up, from almost zero to ~30%

**To Reproduce**
Steps to reproduce the behaviour:
1. Start any game
2. CPU usage for DisplayMagician   jumps up to ~30% while game is running (unwanted)
3. Exit game, and DisplayMagician CPU usage goes down to almost zero (which is wanted)

**Expected behaviour**
CPU usage for DisplayMagician should never sky rocket while monitoring for game to exit.

**Screenshots**
![IMG_20220517_210242__01__01](https://user-images.githubusercontent.com/6009115/168894657-4d785bbc-75a6-4c19-ae4a-81af414bd4cb.jpg)

**Enviroment:**
 - Windows Version:  Win10
 - DisplayMagician Version 2.3.1.60
 - Video Card: Asus 980TI  + iGPU (Intel)
 - Number and make/model of monitors: Nvidia Surround + extended on iGPU.

*Logs*:
[DisplayMagician-Support-20220517-2118.zip](https://github.com/terrymacdonald/DisplayMagician/files/8711054/DisplayMagician-Support-20220517-2118.zip)

**Comment by terrymacdonald** [comment]
Hmmm. This is indeed news to me. I wonder if the app detection logic is not sleeping long enough.

I will have to do some research myself on my system to see if it's something that is experienced in multiple places, or whether something else is causing that to happen just on your machine. 

I'm a bit time limited over the next few weeks, but I'll try to get to it as soon as I can.

Thanks
Terry

**Comment by mattno** [comment]
From the logs I can see

```
2022-05-17 20:08:01.2621|INFO|DisplayMagician.ShortcutRepository|ShortcutRepository/RunShortcut: Starting the game that we wanted to run, and that we're going to monitor and watch
2022-05-17 20:08:01.2621|INFO|DisplayMagician.ShortcutRepository|ShortcutRepository/RunShortcut: Starting the rFactor 2 Steam Game, and then we're going to monitor it to wait for it to close.
2022-05-17 20:08:01.5493|TRACE|DisplayMagician.Processes.ProcessUtils|ProcessUtils/TryExecute: steam://rungameid/365960 was started successfully.
2022-05-17 20:08:01.5493|TRACE|DisplayMagician.Processes.ProcessUtils|ProcessUtils/StartProcess: steam://rungameid/365960  has successfully been started by TryExecute
2022-05-17 20:08:03.0545|DEBUG|DisplayMagician.ShortcutRepository|ShortcutRepository/RunShortcut: Found at least one GameLibrary process has started
2022-05-17 20:08:03.0545|DEBUG|DisplayMagician.ShortcutRepository|ShortcutRepository/RunShortcut: Pausing to let the game library start the game.
2022-05-17 20:08:08.0906|DEBUG|DisplayMagician.ShortcutRepository|ShortcutRepository/RunShortcut: Found the 'rFactor 2' process has started
2022-05-17 20:08:08.0906|DEBUG|DisplayMagician.ShortcutRepository|ShortcutRepository/RunShortcut: waiting for Steam Game rFactor 2 to exit.
2022-05-17 21:14:02.2596|DEBUG|DisplayMagician.ShortcutRepository|ShortcutRepository/RunShortcut: Steam Game rFactor 2 is no longer running (IsRunning is false).
2022-05-17 21:14:02.2596|DEBUG|DisplayMagician.ShortcutRepository|ShortcutRepository/RunShortcut: Steam Game rFactor 2 has exited.
```

The "high" CPU consumption is then seen  `waiting for Steam Game rFactor 2 to exit`  (I guess)

**Comment by terrymacdonald** [comment]
Yeah, that's the bit I was going to look at. I need to see what in that
loop is taking CPU time.

Thanks
Terry

**Comment by terrymacdonald** [comment]
Hi @mattno,

Good news. I found out that I'd used a Task.Delay where I should have used a Thread.Sleep. I tested on my local system and found that the CPU usage dropped from 52% to 0.1%, so I'm fairly confident this was the cause of the issue.

I'm going to close this issue as it is fixed in the code, and the bugfix will be included in a 2.3.2 hotfix. At this stage the 2.3.2 is due within the next month, as there are some other complicated fixes that are due for release in 2.3.2.

Thanks
Terry
